bindgen/src/gen_go/mod.rs

```diff
@@ -475,6 +475,21 @@ pub mod filters {
         Ok(result)
     }
 
+    // Return the Go package name for the given type, if it has one, followed by a '.'
+    // Returns the empty string if the type has no package name associated with it e.g primitive types.
+    //
+    // for multi-package bindings, it may be required to specify which
+    // RustBufferI should be used when creating completeFunc. Failure to
+    // specify the right package here will result in compilation errors.
+    pub fn maybe_namespace(type_: &impl AsType) -> Result<String, askama::Error> {
```

Comment:
I would implement this as part of `ffi_type_name` function itself, that would be more robust.
```diff
diff --git a/bindgen/src/gen_go/mod.rs b/bindgen/src/gen_go/mod.rs
index 1c1c638..9781ea7 100644
--- a/bindgen/src/gen_go/mod.rs
+++ b/bindgen/src/gen_go/mod.rs
@@ -465,11 +465,14 @@ pub mod filters {
     }
 
     /// FFI type name to be used to reference cgo types
-    pub fn ffi_type_name<T: Clone + Into<FfiType>>(type_: &T) -> Result<String, askama::Error> {
+    pub fn ffi_type_name(type_: &Type) -> Result<String, askama::Error> {
         let ffi_type: FfiType = type_.clone().into();
         let result = match ffi_type {
             FfiType::RustArcPtr(_) => "unsafe.Pointer".into(),
-            FfiType::RustBuffer(_) => "RustBufferI".into(),
+            FfiType::RustBuffer(_) => match type_ {
+                Type::External { namespace, .. } => format!("{}.RustBufferI", namespace),
+                _ => "RustBufferI".into(),
+            },
             _ => format!("C.{}", oracle().ffi_type_label(&ffi_type)),
         };
         Ok(result)
```